cli/cmd/uninstall/uninstall.go

```diff
@@ -0,0 +1,439 @@
+			return 1
+		} else {
+			c.UI.Output("Successfully uninstalled Consul Helm release", terminal.WithSuccessStyle())
+		}
+	}
+
+	if c.flagSkipWipeData {
+		c.UI.Output("Skipping deleting PVCs, secrets, and service accounts.", terminal.WithSuccessStyle())
+		return 0
+	}
+
+	// At this point, even if no Helm release was found and uninstalled, there could
+	// still be PVCs, Secrets, and Service Accounts left behind from a previous installation.
+	// If there isn't a foundReleaseName and foundReleaseNamespace, we'll use the values of the
+	// flags c.flagReleaseName and c.flagNamespace. If those are empty we'll prompt the user that
+	// those should be provided to fully clean up the installation.
+	if !found {
+		if c.flagReleaseName == "" || c.flagNamespace == "" {
+			c.UI.Output("No existing Consul Helm installation was found. To search for existing PVCs, secrets, and service accounts left behind by a Consul installation, please provide -release-name and -namespace.", terminal.WithInfoStyle())
+			return 0
+		}
+		foundReleaseName = c.flagReleaseName
+		foundReleaseNamespace = c.flagNamespace
+	}
+
+	// Prompt with a warning for approval before deleting PVCs, Secrets and ServiceAccounts. If flagWipeData is true,
+	// then it will proceed to delete those without a prompt.
+	if !c.flagWipeData {
+		confirmation, err := c.UI.Input(&terminal.Input{
+			Prompt: "WARNING: Proceed with deleting PVCs, Secrets, and ServiceAccounts? \n Only approve if all data from previous installation can be deleted (y/n)",
+			Style:  terminal.WarningStyle,
+			Secret: false,
+		})
+		if err != nil {
+			c.UI.Output(err.Error(), terminal.WithErrorStyle())
+			return 1
+		}
+		confirmation = strings.TrimSuffix(confirmation, "\n")
+		if !(strings.ToLower(confirmation) == "y" || strings.ToLower(confirmation) == "yes") {
+			c.UI.Output("Uninstall aborted without deleting PVCs, Secrets, and ServiceAccounts.", terminal.WithInfoStyle())
+			return 1
+		}
+	}
+
+	err = c.deletePVCs(foundReleaseName, foundReleaseNamespace)
+	if err != nil {
+		c.UI.Output(err.Error(), terminal.WithErrorStyle())
+		return 1
+	}
+
+	err = c.deleteSecrets(foundReleaseName, foundReleaseNamespace)
+	if err != nil {
+		c.UI.Output(err.Error(), terminal.WithErrorStyle())
+		return 1
+	}
+
+	err = c.deleteServiceAccounts(foundReleaseName, foundReleaseNamespace)
+	if err != nil {
+		c.UI.Output(err.Error(), terminal.WithErrorStyle())
+		return 1
+	}
+
+	return 0
+}
+
+func (c *Command) Help() string {
+	c.once.Do(c.init)
+	s := "Usage: kubectl consul uninstall [options]" + "\n\n" + "Uninstall Consul and clean up all data." + "\n" +
+		"Any data store in Consul will not be recoverable." + "\n" + c.help
+	return s
+}
+
+func (c *Command) Synopsis() string {
+	return "Uninstall Consul deployment."
+}
+
+const help = `
+Usage: kubectl consul uninstall [options]
+
+  Uninstall Consul and clean up all data.
+  This is a destructive action. Any data store in Consul will
```

Comment:
```suggestion
  This is a destructive action. Any data stored in Consul will
```

Reply:
Ended up removing this since we won't always remove all the data